### ISSUE TITLE
shouldThrow doesn't check the exact class of the exception

### ISSUE THREAD

**Issue by joegreen88** [phpspec/phpspec#1030]
I want to check the class and message of an exception that is thrown during one of my specs. 

So I write my spec like this:

    function it_throws_exception_for_invalid_status()
    {
        $this->beConstructedWith("hello");
        $e = new \DomainException("'hello' is not an allowed value. Allowed values are: completed, failed");
        $this->shouldThrow($e)->duringInstantiation();
    }

Okay great, phpspec detects the exception. Good job.

Now a couple of days later I decide that I want to throw a custom exception class instead of a `\DomainException` in my code. So I update the code and throw a new type of exception that extends `\DomainException`. I run phpspec expecting to see my test fail, but it passes.

Long story short: shouldThrow() doesn't check the exact class of the exception, but rather it does an instanceof type check.

I don't know if this is intentional or not; it's undocumented either way so I would probably expect my spec to fail in these circumstances.

Best,
Joe

**Comment by ciaranmcnulty** [comment]
This is intended, but the documentation should reflect it better.

Your spec says the object should throw a DomainException and indeed it does (albeit a more specific DomainException).
